openpype/hosts/houdini/plugins/publish/collect_rop_frame_range.py

```diff
@@ -2,40 +2,94 @@
 """Collector plugin for frames data on ROP instances."""
 import hou  # noqa
 import pyblish.api
+from openpype.lib import BoolDef
 from openpype.hosts.houdini.api import lib
+from openpype.pipeline import OptionalPyblishPluginMixin
 
 
-class CollectRopFrameRange(pyblish.api.InstancePlugin):
+class CollectRopFrameRange(pyblish.api.InstancePlugin,
+                           OptionalPyblishPluginMixin):
+
     """Collect all frames which would be saved from the ROP nodes"""
 
+    hosts = ["houdini"]
     order = pyblish.api.CollectorOrder
     label = "Collect RopNode Frame Range"
+    use_asset_handles = True
 
     def process(self, instance):
 
         node_path = instance.data.get("instance_node")
         if node_path is None:
             # Instance without instance node like a workfile instance
+            self.log.debug(
+                "No instance node found for instance: {}".format(instance)
+            )
             return
 
         ropnode = hou.node(node_path)
-        frame_data = lib.get_frame_data(ropnode)
+        asset_data = instance.context.data["assetEntity"]["data"]
+
+        attr_values = self.get_attr_values_from_data(instance.data)
+        if not attr_values.get("use_handles"):
+            asset_data["handleStart"] = 0
+            asset_data["handleEnd"] = 0
+
+        frame_data = lib.get_frame_data(ropnode, asset_data, self.log)
```

Comment:
```suggestion
        attr_values = self.get_attr_values_from_data(instance.data)
        if attr_values.get("use_handles", self.use_asset_handles):
            asset_data = instance.context.data["assetEntity"]["data"]
            handle_start = asset_data.get("handleStart", 0)
            handle_end = asset_data.get("handleEnd", 0)
        else:
            handle_start = 0
            handle_end = 0

        frame_data = lib.get_frame_data(ropnode, handle_start, handle_end, self.log)
```
Would become this if we refactored the `asset_data` argument to explicitly refer to the handles instead to clarify the function.